pkg/controllers/networkPolicy/policy_controller.go

```diff
@@ -0,0 +1,261 @@
+package networkPolicy
+
+import (
+	"reflect"
+	"strings"
+	"time"
+	log "github.com/sirupsen/logrus"
+	calicocache "github.com/projectcalico/k8s-policy/pkg/cache"
+	"github.com/projectcalico/k8s-policy/pkg/controllers/controller"
+	"github.com/projectcalico/k8s-policy/pkg/converter"
+	"github.com/projectcalico/libcalico-go/lib/api"
+	"github.com/projectcalico/libcalico-go/lib/client"
+	"github.com/projectcalico/libcalico-go/lib/errors"
+	"k8s.io/apimachinery/pkg/fields"
+	uruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/tools/cache"
+	k8sNetV1 "k8s.io/kubernetes/pkg/apis/extensions"
+)
+
+// PolicyController Implements Controller interface
+// Responsible for monitoring kubernetes network policies and
+// syncing them to Calico datastore.
+type PolicyController struct {
+	indexer        cache.Indexer
+	informer       cache.Controller
+	calicoObjCache calicocache.ResourceCache
+	calicoClient   *client.Client
+	k8sClientset   *kubernetes.Clientset
+}
+
+// NewPolicyController Constructor for PolicyController
+func NewPolicyController(k8sClientset *kubernetes.Clientset, calicoClient *client.Client) controller.Controller {
+	policyConverter := converter.NewPolicyConverter()
+
+	// Function returns map of policyName:policy stored by policy controller
+	// on ETCD datastore.
+	listFunc := func() (map[string]interface{}, error) {
+
+		npMap := make(map[string]interface{})
+
+		// Get all policies from ETCD datastore
+		calicoPolicies, err := calicoClient.Policy().List(api.PolicyMetadata{})
+		if err != nil {
+			return npMap, err
+		}
+
+		// Filter out only objects that are written by policy controller
+		for _, policy := range calicoPolicies.Items {
```

Comment:
I think the best we can do is use the `.` separator as an indicator.  It's not great, but needed for a smooth upgrade.

The old controller wrote policies with the name format of `namespace.name`, so we can look for that.

We should decide on the right format for the new policy controller to write the policies in though.  

CC @robbrockbank